bundles/org.openhab.core.io.rest.swagger/src/main/java/org/openhab/core/io/rest/swagger/impl/RESTResource.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.io.rest.swagger.impl;
+
+import java.io.IOException;
+import java.util.Map;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.core.io.rest.RESTConstants;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ServiceScope;
+import org.osgi.service.jaxrs.whiteboard.JaxrsWhiteboardConstants;
+import org.osgi.service.jaxrs.whiteboard.propertytypes.JSONRequired;
+import org.osgi.service.jaxrs.whiteboard.propertytypes.JaxrsApplicationSelect;
+import org.osgi.service.jaxrs.whiteboard.propertytypes.JaxrsResource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import de.maggu2810.jaxrswb.swagger1.gen.JaxRsWhiteboardSwaggerGenerator;
+
+/**
+ * An endpoint to generate and provide a Swagger 1 description.
+ *
+ * @author Markus Rathgeb - Initial contribution
+ */
+@Component(service = { RESTResource.class }, scope = ServiceScope.PROTOTYPE)
```

Comment:
Do we need the `ServiceScope` here? You removed it everywhere else.

```suggestion
@Component(service = RESTResource.class)
```

Reply:
It's explained in the whiteboard [resource lifecycle](https://docs.osgi.org/specification/osgi.cmpn/7.0.0/service.jaxrs.html#d0e133288). To me it seems that using a prototype scope may be overkill and also causes more overhead.

Reply:
Everything still seems to work without it so I'll remove it here as well.